crates/rspack_plugin_copy/src/lib.rs

```diff
@@ -244,6 +256,38 @@ impl CopyRspackPlugin {
       }
     };
 
+    if let Some(transform) = &pattern.transform {
+      if let Some(absolute_filename) = absolute_filename.to_str() {
+        match transform {
+          Transformer::Fn(transformer) => {
+            let transformed = transformer(
+              match &source {
+                RawSource::Source(code) => code.to_owned(),
+                RawSource::Buffer(buffer) => String::from_utf8(buffer.to_owned())
+                  .unwrap_or_else(|err| panic!("Transform buffer source error: {err}")),
```

Comment:
What if the `source` is a binary here? IIRC, `copy-webpack-plugin` transforms both `string` and `Buffer`.